server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1773,39 +1782,121 @@ public File exportData(
+        " will contain the href to download the generated file.",
+        value = "Async Consumer Export (manifest)",
+        response = JobDetail.class)
+    @ApiResponses({ @ApiResponse(code = 403, message = ""), @ApiResponse(code = 500, message = ""),
+        @ApiResponse(code = 404, message = "") })
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{consumer_uuid}/export/async")
+    public JobDetail exportDataAsync(
+        @Context HttpServletResponse response,
+        @PathParam("consumer_uuid") @Verify(Consumer.class) String consumerUuid,
+        @QueryParam("cdn_label") String cdnLabel,
+        @QueryParam("webapp_prefix") String webAppPrefix,
+        @QueryParam("api_url") String apiUrl) {
+        Consumer consumer = verifyExportParams(consumerUuid, cdnLabel);
+        return manifestManager.generateManifestAsync(consumer, cdnLabel, webAppPrefix, apiUrl);
+    }
 
-        File archive;
-        try {
-            archive = exporter.getFullExport(consumer, cdnLabel, webAppPrefix, apiUrl);
-            response.addHeader("Content-Disposition", "attachment; filename=" +
-                archive.getName());
+    /**
+     * Downloads an asynchronously generated consumer export file (manifest). If the file
+     * was successfully downloaded, it will be deleted.
+     *
+     * @param response
+     * @param consumerUuid the UUID of the target consumer.
+     * @param exportId the id of the stored export.
+     */
+    @ApiOperation(
+        notes = "Downloads an asynchronously generated consumer export file (manifest).",
+        value = "Async Consumer Export (manifest) Download",
+        response = File.class)
+    @ApiResponses({ @ApiResponse(code = 403, message = ""), @ApiResponse(code = 500, message = ""),
+        @ApiResponse(code = 404, message = "") })
+    @GET
+    @Produces("application/zip")
+    @Path("{consumer_uuid}/export/{export_id}")
+    public void downloadExistingExport(
+        @Context HttpServletResponse response,
+        @PathParam("consumer_uuid") @Verify(Consumer.class) String consumerUuid,
+        @PathParam("export_id") String exportId) {
 
-            sink.queueEvent(eventFactory.exportCreated(consumer));
-            return archive;
-        }
-        catch (ExportCreationException e) {
-            throw new IseException(i18n.tr("Unable to create export archive"),
-                e);
+        // *******************************************************************************
+        // NOTE: If changing the path or parameters of this end point, be sure to update
+        // the HREF generation in ConsumerResource.buildAsyncDownloadManifestHref.
+        // *******************************************************************************
+
+        Consumer consumer = consumerCurator.verifyAndLookupConsumer(consumerUuid);
+        if (consumer.getType() == null || !consumer.getType().isManifest()) {
```

Comment:
There are other places we perform the same validation. any scope for reducing redundancy?